### PR TITLE
Merge a number of workers together.

### PR DESCRIPTION
**Pull request by erikjohnston**
The aim here is to only have one worker app so that we can simplify things a bit.

This is quite a dumb copy and paste from the worker files into a new `generic_worker` app. This can be cleaned up a lot, but that is for a separate PR. Currently, nothing changes in terms of running workers, again that is going to be a separate PR to support that and updating the docs.

### CODE REVIEW

File: tests/app/test_frontend_proxy.py [hunk 1]
```diff
@@ -46,9 +46,7 @@ def test_listen_http_with_presence_enabled(self):
         # Grab the resource from the site that was told to listen
         self.assertEqual(len(self.reactor.tcpServers), 1)
         site = self.reactor.tcpServers[0][1]
-        self.resource = (
-            site.resource.children[b"_matrix"].children[b"client"].children[b"r0"]
-        )
+        self.resource = site.resource.children[b"_matrix"].children[b"client"]
```
Comment (anoadragon453) [inline tests/app/test_frontend_proxy.py]: Should this change be reflected somewhere else? Seems odd to just lop off `r0` here.

Comment (erikjohnston) [reply tests/app/test_frontend_proxy.py]: This is just a side effect of whether we register the `JsonResource` at `/_matrix/client/r0` or `/_matrix/client` in the code. For whatever reason the frontend proxy was registering the resource at `/_matrix/client/r0`, while everything else uses `/_matrix/client`

File: synapse/app/user_dir.py [hunk 2]
```diff
@@ -1,6 +1,6 @@
 #!/usr/bin/env python
 # -*- coding: utf-8 -*-
-# Copyright 2017 Vector Creations Ltd
+# Copyright 2016 OpenMarket Ltd
```
[the remark this answers is not in the document]
Comment (erikjohnston) [reply synapse/app/user_dir.py]: Whoops, didn't mean to change this copyright. I've reverted and updated the copyright on generic worker too, which I think is the right thing.

File: tests/app/test_frontend_proxy.py (hunk 1 above)
Comment (anoadragon453) [reply tests/app/test_frontend_proxy.py]: Ah ok, so just fixing an inconsistency. Cool :)

**Comment by anoadragon453** [comment]
Looks like a flaky test for: `Guest user can set display names`

**Pull request merged**